src/components/inline_edit/inline_edit_form.tsx

```diff
@@ -186,20 +223,21 @@ export const EuiInlineEditForm: FunctionComponent<EuiInlineEditFormProps> = ({
 
   const cancelInlineEdit = () => {
     setEditModeValue(readModeValue);
+    onCancel && onCancel(readModeValue as string);
     setIsEditing(false);
     requestAnimationFrame(() => readModeFocusRef.current?.focus());
   };
 
   const saveInlineEditValue = async () => {
     // If an onSave callback is present, and returns false, stay in edit mode
     if (onSave) {
-      const onSaveReturn = onSave(editModeValue);
+      const onSaveReturn = onSave(value as string);
```

Comment:
Rather than casting `as string` here, we can trick Typescript into understanding all values will be strings in a different way (see next comment)

```suggestion
      const onSaveReturn = onSave(value);
```